pkg/ee/cluster-backup/controller.go

```diff
@@ -0,0 +1,243 @@
+		}
+		return reconcile.Result{}, err
+	}
+
+	// Resource is marked for deletion.
+	if cluster.DeletionTimestamp != nil {
+		return reconcile.Result{}, nil
+	}
+
+	backupConfig, err := clusterutil.FetchClusterBackupConfigWithSeedClient(ctx, r.Client, cluster, r.log)
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to fetch cluster backup config: %w", err)
+	}
+
+	// clusterbackup is disabled. Nothing to do.
+	if !backupConfig.Enabled {
+		return reconcile.Result{}, nil
+	}
+
+	// Add a wrapping here so we can emit an event on error
+	result, err := kubermaticv1helper.ClusterReconcileWrapper(
+		ctx,
+		r.Client,
+		r.workerName,
+		cluster,
+		r.versions,
+		kubermaticv1.ClusterConditionClusterBackupControllerReconcilingSuccess,
+		func() (*reconcile.Result, error) {
+			return r.reconcile(ctx, cluster, backupConfig)
+		},
+	)
+
+	if result == nil || err != nil {
+		result = &reconcile.Result{}
+	}
+
+	if err != nil {
+		r.recorder.Event(cluster, corev1.EventTypeWarning, "ReconcilingError", err.Error())
+	}
+
+	return *result, err
+}
+
+func (r *reconciler) reconcile(ctx context.Context, cluster *kubermaticv1.Cluster, backupConfig *resources.ClusterBackupConfig) (*reconcile.Result, error) {
+	if err := r.ensureClusterBackupUserClusterResources(ctx, cluster, backupConfig); err != nil {
+		return nil, fmt.Errorf("failed to ensure Cluster Backup user-cluster resources: %w", err)
+	}
+
+	if err := r.ensureClusterBackupSeedClusterResources(ctx, cluster, backupConfig); err != nil {
+		return nil, fmt.Errorf("failed to ensure Cluster Backup Seed cluster resources: %w", err)
+	}
+	return nil, nil
+}
+
+func (r *reconciler) ensureClusterBackupUserClusterResources(ctx context.Context, cluster *kubermaticv1.Cluster, backupConfig *resources.ClusterBackupConfig) error {
+	userClusterClient, err := r.userClusterConnectionProvider.GetClient(ctx, cluster)
+	if err != nil {
+		return fmt.Errorf("failed to get user cluster client: %w", err)
+	}
+	nsReconcilers := []reconciling.NamedNamespaceReconcilerFactory{
+		userclusterresources.NamespaceReconciler(),
+	}
+	if err := reconciling.ReconcileNamespaces(ctx, nsReconcilers, "", userClusterClient); err != nil {
+		return fmt.Errorf("failed to reconcile velero namespace: %w", err)
+	}
+
+	saReconcilers := []reconciling.NamedServiceAccountReconcilerFactory{
+		userclusterresources.ServiceAccountReconciler(),
+	}
+	if err := reconciling.ReconcileServiceAccounts(ctx, saReconcilers, resources.ClusterBackupNamespaceName, userClusterClient); err != nil {
+		return fmt.Errorf("failed to reconcile velero service account: %w", err)
+	}
+
+	clusterRoleBindingReconciler := []reconciling.NamedClusterRoleBindingReconcilerFactory{
+		userclusterresources.ClusterRoleBindingReconciler(),
+	}
+
+	if err := reconciling.ReconcileClusterRoleBindings(ctx, clusterRoleBindingReconciler, "", userClusterClient); err != nil {
+		return fmt.Errorf("failed to reconcile velero cluster role binding: %w", err)
```

Comment:
please make up your mind whether you write resource names in "lower case name" or "UpperCaseName". This function alone has both "cluster role binding", but also "CustomResourceDefinition" ,)

Also, it's Velero, not velero.

And "Cluster Backup" should probably be "cluster backup". ^^